include/sys/emmc.h

```diff
@@ -0,0 +1,303 @@
+#ifndef _SYS_EMMC_H_
+#define _SYS_EMMC_H_
+
+#include <inttypes.h>
+#include <sys/mimiker.h>
+#include <sys/device.h>
+#include <sys/endian.h>
+#include <stddef.h>
+
+/* These operation modes are described in JESD86-A441, with a brief summary
+   at page 27 */
+typedef enum emmc_op_mode {
+  EMMCMODE_BOOT,
+  EMMCMODE_CARD_ID,
+  EMMCMODE_INTR,
+  EMMCMODE_TRANSFER,
+  EMMCMODE_INA,
+} emmc_op_mode_t;
+
+typedef enum emmc_result {
+  EMMC_OK,
+  EMMC_ERR_IERR,
+  EMMC_ERR_TIMEOUT,
+  EMMC_ERR_NOT_RESPONDING,
+  EMMC_ERR_INTR,
+  EMMC_ERR_BUSY,
+} emmc_result_t;
+
+/* Response types R1-R5 are described in JESD86-A441, page 94.
+ * Response R1 and R1b are normally both represented as `EMMCRESP_R1`,
+ * because they are structurally identical. */
+typedef enum emmc_resp_type {
+  EMMCRESP_NONE,
+  EMMCRESP_R1,
+  EMMCRESP_R1B, /* Can be used only as an argument for `emmc_send_app_cmd` */
```

Comment:
Why does it have letter "B"? Is it supposed to note that it's "second R1"? 
If it's an R1 only for `emmc_send_app_cmd` (as it seems to be) then, I think, it should have it or part of it in name. So the name of response type tells me so, I don't have to read comment. It's easier then to not use that R1B for anything else than `emmc_send_app_cmd`

Reply:
These names (including "**R1b**") are literally how the specification calls these responses. It also states that **R1** and **R1b** are structurally the same and the difference comes to the way the controller received them, which is irrelevant for a developer, unless they want to instruct the controller on how to handle a command that is not in specs.

The comment should be changed this. It should not state an intended use, as there's only one or for `emmc_resp_type_t` as a function parameter in this interface, but instead it should say that commands which return either **R1** or **R1b** responses are all reported to drivers as returning **R1**. Perhaps this comment could be made above `emmc_resp_type` declaration.

Reply:
Okay, understood.